.github/workflows/release.yml

```diff
@@ -0,0 +1,58 @@
+name: aiodynamo release workflow
+
+on:
+  release:
+    types: [released] # only triggered on release, not pre-release or draft release
+
+permissions:
+  contents: read
+  id-token: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  PYTHON_VERSION_MIN: 3.8
+  TARGET_BRANCH: ${{ github.event.release.target_commitish }}
+  DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
+  TAG_NAME: ${{ github.ref_name }}
+
+jobs:
+  release:
+    name: Release to PyPI
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check whether release target branch is default branch
+        if: env.TARGET_BRANCH != env.DEFAULT_BRANCH
+        uses: actions/github-script@v6
+        with:
+          script: |
+            core.setFailed('Target branch (${{ env.TARGET_BRANCH }}) is not default branch (${{ env.DEFAULT_BRANCH }}).')
+      
+      - uses: actions/checkout@v3
+      
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION_MIN }}
+          
+      - run: curl -sSL https://install.python-poetry.org | python3 -
```

Comment:
Minor:
There's a github action available for that, which allows pinning poetry version.

Reply:
We can also pin the version here with
```
curl -sSL https://install.python-poetry.org | POETRY_VERSION=X.X.X python3 -
```

I just prefer to use the official installation command rather than depend on some third party action, especially since it's only a one liner.

Reply:
And about the version, I just thought using the latest poetry version should be fine for this.